src/utils/utils.rs

```diff
@@ -593,6 +593,25 @@ pub fn toolchain_sort<T: AsRef<str>>(v: &mut Vec<T>) {
     });
 }
 
+fn copy_and_delete<'a, N>(
+    name: &'static str,
+    src: &'a Path,
+    dest: &'a Path,
+    notify_handler: &'a dyn Fn(N),
+) -> Result<()>
+where
+    N: From<Notification<'a>>,
+{
+    // https://github.com/rust-lang/rustup/issues/1239
+    // This uses std::fs::copy() instead of the faster std::fs::rename() to
+    // avoid cross-device link errors.
+    if src.is_file() {
```

Comment:
rather than checking is_file, check is_dir(); that way you'll correctly fall back to a simple copy for symlinks